go/test/endtoend/cluster/cluster_process.go

```diff
@@ -79,6 +83,12 @@ type LocalProcessCluster struct {
 	VtctldExtraArgs []string
 
 	EnableSemiSync bool
+
+	mx                *sync.Mutex
```

Comment:
The purpose of this mutex is to ensure that Teardown gets executed exactly once, is that correct?
Can you either rename it to be more descriptive, or add a comment?